src/engine/nodes/OSCSenderNode.cpp

```diff
@@ -100,16 +100,18 @@ void OSCSenderNode::run ()
         MidiMessage msg;
         int frame;
 
-        ScopedLock sl (lock);
-
         while (iter1.getNextEvent (msg, frame))
         {
             OSCMessage oscMsg = Util::processMidiToOscMessage (msg);
             oscSender.send (oscMsg);
 
-            if (oscMessagesToLog.size() > maxOscMessages)
-                oscMessagesToLog.erase ( oscMessagesToLog.begin() );
-            oscMessagesToLog.push_back ( oscMsg );
+            {
+                ScopedLock sl (lock);
+
+                if (oscMessagesToLog.size() > maxOscMessages)
+                    oscMessagesToLog.erase ( oscMessagesToLog.begin() );
+                oscMessagesToLog.push_back ( oscMsg );
+            }
         }
```

Comment:
@mfisher31 Is this OK to create a scoped lock repeatedly inside a loop? Maybe it's better to send off the out-going OSC messages first, then afterwards lock+loop for the logs?